### ISSUE TITLE
missing option to add withCredentials to the upload files onto a secure server

### ISSUE THREAD

**Issue by pattabhi551** [vaadin/vaadin-upload#133]
The upload component creates its own xhr requests to upload the files to the target location, but misses the attribute to configure the withCredentials option to the xhr request and makes it impossible to use the component to upload files onto a secure server. 

A new property to configure the xhr to have withCredentials to true (like iron-ajax does) would be quite handy as currently uploading to secure endpoint throw a 403 in spite of having the necessary cookies.

**Comment by platosha** [comment]
Hi! Although there is no `withCredentials` property in the upload itself, this is still possible. You can use the `upload-request` event to add custom configuration for the XHR before it is sent. The xhr object is passed in the event detail. Please try:

```javascript
upload.addEventListener('upload-request', function(event) {
  event.detail.xhr.withCredentials = true;
});
```

Does this meet your needs?

**Comment by manolo** [comment]
Fixed in #218
